opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/RuntimeEnvironment.java

```diff
@@ -125,14 +126,16 @@
     private RuntimeEnvironment() {
         configuration = new Configuration();
         configLock = new ReentrantReadWriteLock();
+        authFrameworkLock = new ReentrantLock();
```

Comment:
i think with this you can just use new Object() and synchronize on that object

Reply:
well, yes. The advantage is that the `synchronized` block looks nicer however using plain `Object` seems strange to me. Is this a fairly common pattern ?

Reply:
Also, could there be some unintended side effects ? As I understand it the `synchronized` will use in-object lock. Is there some other operation that could use the same lock ?

Reply:
you mean like in
https://github.com/oracle/opengrok/blob/ff44f24ac10337e272510634599e287cb25d0e9c/opengrok-indexer/src/main/java/org/opengrok/indexer/web/messages/MessagesContainer.java#L64 or
https://github.com/oracle/opengrok/blob/ff44f24ac10337e272510634599e287cb25d0e9c/opengrok-indexer/src/main/java/org/opengrok/indexer/history/FileHistoryCache.java#L71
?

object is the simplest reference and that's all you need to synchronize

Reply:
there's no side effect, it is the same like using `this` but instead you use custom object